Adding Scenario Documentation Policy and RACI Chart, additional documenation cleanup (#6)
CSPI-QE/ocp-ci-docs: docs/Policy/Documentation/Scenario_Documentation_Policy.md

# Scenario Documentation Policy<!-- omit from toc -->

## Table of Contents<!-- omit from toc -->
- [Introduction](#introduction)
- [How to Document a Scenario](#how-to-document-a-scenario)
  - [Getting Started](#getting-started)
  - [Adding Documentation](#adding-documentation)
    - [General Information](#general-information)
    - [Purpose](#purpose)
    - [Process](#process)
    - [Custom Images](#custom-images)
- [Conclusion](#conclusion)

## Introduction

> Documenting our scenarios is an important part of the CSPI onboarding and maintenance process. Ensuring we have a comprehensive document for each scenario will help future current and future team members debug and maintain each scenario. Please follow this documentation policy to write helpful documentation for each new scenario.

Author: ```suggestion
Documenting our scenarios is an important part of the CSPI onboarding and maintenance process. Ensuring we have a comprehensive document for each scenario will help current and future team members debug and maintain each scenario. Please follow this documentation policy to write helpful documentation for each new scenario.
```

## How to Document a Scenario
Each scenario written by CSPI within OpenShift CI should have a `README.md` file associated with it in the same folder as the scenario's configuration file. For example the MTR scenario's documentation can be found at `ci-operator/config/windup/windup_integration_test/README.md` along with the configuration YAML file. Documentation should be written in Markdown. If you are unfamiliar with Markdown, please use [this list of resources](Markdown_Resources.md) to familiarize yourself with it.

### Getting Started

> 1. Create the `README.md` file in the same director that holds the scenario's configuration YAML file.

Author: ```suggestion
1. Create the `README.md` file in the same directory that holds the scenario's configuration YAML file.
```

   - **Note:** This file should be in the `ci-operator/config/{test-org}/{test-repo}/` directory within the [openshift/release](https://github.com/openshift/release) repository.
2. Add a level-1 header to the top of the file with the name of the scenario
   - **Note:** Please add `<!-- omit from toc -->` to the end of the header to avoid adding it to the Table of Contents.
   - **Example:** `# windup-windup_integration_test-main<!-- omit from toc -->`
3. Add a level-2 header below the first header with the text "Table of Contents"
   - **Example:** `## Table of Contents <!-- omit from toc -->`
4. Add the Table of Contents
   1. In Visual Studio Code, make sure the [Markdown All in One](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) plugin is installed
   2. Navigate to the line below the "Table of Contents" header and press `CTRL+SHIFT+P` to bring up the Command Pallette.
   3. Type in `Markdown All in One: Create Table of Contents`
   4. Press enter
   - **Note:** This plugin will create the Table of Contents for you and keep it automatically updated with hyperlinks upon every save of the document. 

### Adding Documentation

> The outline of these documents should generally follow the following structure:

Reviewer: ```suggestion
The outline of these documents should generally follow the structure:
```

Author: Fixed

#### General Information
This section of the documentation should include the following items:
- A link to the test repository
- Which layered product(s) is being tested
- Who maintains this scenario?

Please feel free to add any additional information or links that may be helpful in this section as well.

**Example:**

<sub><sup>`ci-operator/config/windup/windup_integration_test/README.md`</sup></sub>
```markdown
## General Information
- **Repository**: [windup/windup_integration_test](https://github.com/windup/windup_integration_test.git)
- **Operator Tested**: [MTR (Migration Toolkit for Runtimes)](https://developers.redhat.com/products/mtr/overview)
- **Maintainers**: CSPI QE
```

#### Purpose
This section of the documentation should outline the purpose of this scenario. While it may seem obvious to us, OpenShift CI is used by many teams and they may stumble across this scenario. It is important that people outside of CSPI and PQE know what this configuration does and why it is being run.

This section can be as short as a sentence or two:

<sub><sup>`ci-operator/config/windup/windup_integration_test/README.md`</sup></sub>
```markdown
## Purpose

> To provision the necessary infrastructure and using that infrastructure to execute MTR interop tests. The results of theses tests should be reported to the appropriate sources following execution.

Reviewer: ```suggestion
To provision the necessary infrastructure and use that infrastructure to execute MTR interop tests on pre release OCP builds. The results of theses tests will be reported to the appropriate sources following execution.
```

Author: Fixed
